### PR TITLE
dbhash sqldiff sqlite-analyzer sqlite 3.40.1

### PR DESCRIPTION
**Pull request by p-linnane**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
for sqlite on linux build
```
  tclsh /tmp/sqlite-20221228-31551-1alhvzj/sqlite-src-3400100/tool/mkshellc.tcl >shell.c
  /bin/sh: 1: tclsh: not found
```

`rpm` and `tracker` test failures for osx build

```
brew test --retry --verbose rpm
brew test --retry --verbose tracker
```

**Comment by chenrui333** [comment]
I can reproduce `rpm` in the master branch, but `tracker` works for me fine in the master branch build.

File: Formula/sqlite.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Sqlite < Formula
   desc "Command-line interface for SQLite"
   homepage "https://sqlite.org/index.html"
-  url "https://sqlite.org/2022/sqlite-autoconf-3400000.tar.gz"
-  version "3.40.0"
-  sha256 "0333552076d2700c75352256e91c78bf5cd62491589ba0c69aed0a81868980e7"
+  url "https://sqlite.org/2022/sqlite-src-3400100.zip"
```
Comment (cho-m) [inline Formula/sqlite.rb:4]: Any reason why this changed from `sqlite-autoconf` tarball to `sqlite-src` zip?

Comment (p-linnane) [reply Formula/sqlite.rb:4]: Ah wait, I misunderstood. That was already fixed. Let me change that back.

**Comment by fxcoudert** [comment]
`rpm` test failure on all targets. Either rpm need a revision bump, or the test needs to be fixed.

**Comment by SMillerDev** [comment]
The test needs to be fixed, that happens outside of this PR too. I'm giving it a try now.

**Comment by SMillerDev** [comment]
https://github.com/Homebrew/homebrew-core/pull/119398 has the fixes, but doesn't pass brew style

**Comment by p-linnane** [comment]
Rebased with @SMillerDev's `rpm` fixes.

**Comment by iMichka** [comment]
Linux failure is unrelated (GitHub rate limit was hit during a recursive dependency test).

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**